### ISSUE TITLE
Deleting Asset from Media Libary produces error

### ISSUE THREAD

**Issue by Maxdola** [strapi/strapi#13879]
### Required System information

- Node.js version: 16.13.0
- NPM version: 8.10
- Strapi version: 4.3.0
- Database: 8.7.3
- Operating system: Kubernetis

### Describe the bug

Deleting an Asset in the Media Libary produces an Error. But the Image is deleted anyway.

### Steps to reproduce the behavior

(This was tested with the Images stored in S3 using @strapi/provider-upload-aws-s3@4.3.0)

1. Goto Media Libary and Delete an asset.

### Expected behavior

The Assed should be deleted without an Error

### Screenshots

![](https://i.imgur.com/WPBgxeC.png)

### Code snippets

```
[2022-07-28 09:04:52.199] error: Cannot read properties of undefined (reading 'length')
TypeError: Cannot read properties of undefined (reading 'length')
    at Object.deleteMany (/app/node_modules/@strapi/plugin-upload/server/controllers/admin-folder-file.js:41:46)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async returnBodyMiddleware (/app/node_modules/@strapi/strapi/lib/services/server/compose-endpoint.js:52:18)
    at async policiesMiddleware (/app/node_modules/@strapi/strapi/lib/services/server/policy.js:24:5)
    at async /app/node_modules/@strapi/strapi/lib/middlewares/body.js:51:9
    at async /app/node_modules/@strapi/strapi/lib/middlewares/logger.js:22:5
    at async /app/node_modules/@strapi/strapi/lib/middlewares/powered-by.js:16:5
    at async cors (/app/node_modules/@koa/cors/index.js:95:16)
    at async /app/node_modules/@strapi/strapi/lib/middlewares/errors.js:13:7
```

**Comment by petersg83** [comment]
Thank you :)
I think It's a duplicate of https://github.com/strapi/strapi/issues/13870.
The fix has been merged and it will be released in the next patch version (In a bit less than 2 weeks).

I propoes we let this issue open in the meantime so it is more visible for people coming on the repo

**Comment by Maxdola** [comment]
> Thank you :) I think It's a duplicate of #13870. The fix has been merged and it will be released in the next patch version (In a bit less than 2 weeks).
> 
> I propoes we let this issue open in the meantime so it is more visible for people coming on the repo

Oh yes I didn't saw the other Issue because I was only looking for open ones. Thank you.

**Comment by petersg83** [comment]
Yes I understand it was not quite visible ^^

As a workaround you can delete an asset by clicking the delete icon in the edition modal :
![Capture d’écran 2022-07-28 à 12 03 26](https://user-images.githubusercontent.com/11708220/181479582-daa0ec12-5ff5-41d4-8093-2e08292b54d2.png)

Another workaround is to delete an empty folder at the same time as deleting assets

**Comment by Maxdola** [comment]
We found this Issue while Testing in the Development Stage. We obviously won't deploy this to Production and will wait for the patch. In my opinion, it should have been published immediately, since it's a significant bug.

**Comment by imokya** [comment]
I got the same error when trying to bulk delete files

**Comment by petersg83** [comment]
Fix released in 4.3.2 :)
